packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/heatmap.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show listEquals;
+import 'package:flutter/foundation.dart' show immutable;
+import 'package:flutter/material.dart' show Color;
+
+import 'types.dart';
+
+/// Uniquely identifies a [Heatmap] among [GoogleMap] heatmaps.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class HeatmapId extends MapsObjectId<Heatmap> {
+  /// Creates an immutable identifier for a [Heatmap].
+  const HeatmapId(String value) : super(value);
+}
+
+/// Draws a heatmap on the map.
+@immutable
+class Heatmap implements MapsObject<Heatmap> {
+  /// Creates an immutable representation of a [Heatmap] to draw on
+  /// [GoogleMap].
+  const Heatmap({
+    required this.heatmapId,
+    this.data = const <WeightedLatLng>[],
+    this.dissipating = true,
+    this.gradient,
+    this.maxIntensity,
+    // Default is 0.6 on web, 0.7 on Android and iOS.
+    this.opacity = 0.7,
+    this.radius = 20,
+    this.minimumZoomIntensity = 0,
+    this.maximumZoomIntensity = 21,
+  }) : // Docs for iOS say [radius] must be between 10 and 50, but anything
+        // higher than 45 causes EXC_BAD_ACCESS.
+        assert(radius >= 10 && radius <= 45);
+
+  /// Uniquely identifies a [Heatmap].
+  final HeatmapId heatmapId;
+
+  @override
+  HeatmapId get mapsId => heatmapId;
+
+  /// The data points to display.
+  final List<WeightedLatLng> data;
+
+  /// Specifies whether heatmaps dissipate on zoom. By default, the radius of
+  /// influence of a data point is specified by the radius option only. When
+  /// dissipating is disabled, the radius option is interpreted as a radius at
+  /// zoom level 0.
+  ///
+  /// Web only.
```

Comment:
Done

Reply:
I understand the reasoning, but where should this information be instead? Platform implementations don't override anything in here, and I don't think we can expect users to read native documentation to find this information.

Reply:
It could be in the implementation package READMEs, that way the documentation is in the same package that actually controls whether or not the statement is true.

Reply:
I can move it, but I fully expect users of the plugin to create issues before reading documentation that isn't stupid easy to access.

Reply:
I moved these to the READMEs